src/AutoRest.CSharp/MgmtTest/Generation/MgmtBaseTestWriter.cs

```diff
@@ -0,0 +1,773 @@
+
+        public MgmtBaseTestWriter(CodeWriter writer, MgmtTypeProvider provider, BuildContext<MgmtOutputLibrary> context) : base(writer, provider, context)
+        {
+        }
+
+        public void WriteTestDecorator()
+        {
+            _writer.Line($"[RecordedTest]");
+
+            var testModelerConfig = Context.Configuration.MgmtConfiguration.TestModeler;
+            string? ignoreReason = testModelerConfig?.IgnoreReason;
+            if (ignoreReason is not null)
+            {
+                _writer.UseNamespace("NUnit.Framework");
+                _writer.Line($"[Ignore(\"{ignoreReason}\")]");
+            }
+        }
+
+        public static object? ConvertToStringDictionary(object dict)
+        {
+            if (dict is null)
+            {
+                return dict;
+            }
+            Type t = dict.GetType();
+            bool isDict = t.IsGenericType && t.GetGenericTypeDefinition() == typeof(Dictionary<,>);
+            if (!isDict)
+            {
+                return dict;
+            }
+            var ret = new Dictionary<string, object?>();
+            foreach (KeyValuePair<object, object> entry in (IEnumerable< KeyValuePair<object, object>>)dict)
+            {
+                ret.Add(entry.Key.ToString()!, ConvertToStringDictionary(entry.Value));
+            }
+            return ret;
+        }
+
+        public static string FormatResourceId(string resourceId)
+        {
+            if (resourceId.Length == 0)
+            {
+                return "/subscriptions/00000000-0000-0000-0000-000000000000";   // provide a fake resourceId to fullfill the ResourceIdentifier check
+            }
+            if (!resourceId.StartsWith('/'))
+            {
+                resourceId = '/' + resourceId;
+            }
+            resourceId = resourceId.Replace("{", "{{").Replace("}", "}}");
+            var elements = resourceId.Split("/");
+            for (int i = 2; i< elements.Length; i+=2)
+            {
+                if (elements[i-1].ToLower()== "subscriptions")
+                {
+                    Regex regex = new Regex("^{[A-Z0-9]{8}-([A-Z0-9]{4}-){3}[A-Z0-9]{12}}$");
+                    Match match = regex.Match(elements[i]);
+                    if (!match.Success)
+                    {
+                        elements[i] = "00000000-0000-0000-0000-000000000000";
+                    }
+                }
+            }
+            return String.Join("/", elements);
+        }
+
+        public static string GetTaskOrVoid(bool isAsync)
+        {
+            return isAsync ? "Task" : "void";
+        }
+
+        public static ExampleGroup? FindExampleGroup(BuildContext<MgmtOutputLibrary> context, MgmtRestOperation restOperation)
+        {
+            if (restOperation is null)
+                return null;
+            foreach (var exampleGroup in context.CodeModel.TestModel?.MockTest.ExampleGroups ?? Enumerable.Empty<ExampleGroup>())
```

Comment:
I believe if the code can run here, we have guaranteed the TestModel is not null, therefore we do not need the null short cut here